src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -492,50 +492,58 @@ private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax n
             var hasErrors = localSymbol.ScopeBinder
                 .ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
-            BoundBlock block;
-            if (node.Body != null)
-            {
-                block = BindEmbeddedBlock(node.Body, diagnostics);
-            }
-            else if (node.ExpressionBody != null)
+            BoundBlock blockBody = node.Body != null ? BindEmbeddedBlock(node.Body, diagnostics) : null;
+            DiagnosticBag expressionBodyDiagnostics = null;
+            BoundBlock expressionBody = null;
+            if (node.ExpressionBody != null)
             {
-                block = BindExpressionBodyAsBlock(node.ExpressionBody, diagnostics);
+                expressionBodyDiagnostics = node.Body == null ? diagnostics : new DiagnosticBag();
+                expressionBody = BindExpressionBodyAsBlock(node.ExpressionBody, expressionBodyDiagnostics);
             }
-            else
+            else if (node.Body == null && node.ExpressionBody == null)
```

Comment:
>node.ExpressionBody == null [](start = 42, length = 27)

Is this condition ever  false?

Reply:
As written, it should be false if ExpressionBody is null but BlockBody isn't. We could remove the `node.ExpressionBody == null` part, but I'll rewrite this along your recommendations above so it'll be removed.

---
In reply to: [177632742](https://github.com/dotnet/roslyn/pull/25782#discussion_r177632742) [](ancestors = 177632742)